src/middlewared/middlewared/plugins/reporting/netdata_configure.py

```diff
@@ -11,9 +13,20 @@ class ReportingService(Service):
 
     @private
     async def cache_netdata_uid_gid(self):
-        user_obj = await self.middleware.call('user.get_user_obj', {'username': 'netdata'})
-        self.NETDATA_UID = user_obj['pw_uid']
-        self.NETDATA_GID = user_obj['pw_gid']
+        # this method runs in middleware init phase so it's important
+        # to handle errors and log a message instead of crashing. If
+        # we crash here, middleware service won't start
+        try:
+            self.NETDATA_UID = await self.middleware.call('user.get_builtin_user_id', NETDATA_USERNAME)
+        except Exception:
+            self.logger.error('Unexpected failure resolving username %r to uid', NETDATA_USERNAME, exc_info=True)
```

Comment:
Maybe also set `self.NETDATA_UID = -1` (and same for GID) on error because this will be passed to chown.

Reply:
So that `netdata_setup()` doesn't break.

Reply:
done